src/network/mcpe/handler/InGamePacketHandler.php

```diff
@@ -1032,4 +1036,28 @@ public function handleEmote(EmotePacket $packet) : bool{
 		$this->player->emote($packet->getEmoteId());
 		return true;
 	}
+
+	public function handleStructureBlockUpdate(StructureBlockUpdatePacket $packet) : bool{
+		$pos = $packet->blockPosition;
+		$chunkX = $pos->getX() >> Chunk::COORD_BIT_SIZE;
+		$chunkZ = $pos->getZ() >> Chunk::COORD_BIT_SIZE;
+		$world = $this->player->getWorld();
+		if(!$world->isChunkLoaded($chunkX, $chunkZ) || $world->isChunkLocked($chunkX, $chunkZ)){
+			return false;
+		}
+
+		$tile = $world->getTileAt($pos->getX(), $pos->getY(), $pos->getZ());
+		$block = $world->getBlockAt($pos->getX(), $pos->getY(), $pos->getZ());
+		if($tile instanceof TileStructureBlock && $block instanceof StructureBlock && $this->player->canInteract($block->getPosition(), 15)){
+			try{
+				$type = StructureBlockType::fromInt($packet->structureEditorData->structureBlockType);
+				$tile->updateFromPacket($packet);
+			}catch(\ValueError $e){
```

Comment:
Why is this being handled? This is an engine error that could be generated by any defective code.